site/source/docs/porting/simd.rst

```diff
@@ -110,6 +110,8 @@ The following legend is used to highlight the expected performance of various in
 
 Certain intrinsics in the table below are marked "virtual". This means that there does not actually exist a native x86 SSE instruction set opcode to implement them, but native compilers offer the function as a convenience. Different compilers might generate a different instruction sequence for these.
 
+In addition to consulting the tables below, you can turn on diagnostics for slow, emulated functions by defining `__EMSCRIPTEN_SIMD_COMPAT_SLOW`. This will print out warnings if you attempt to use any of the slow paths (corresponding to ❌ or 💣 in the legend).
```

Comment:
We do not currently have any Emscripten compiler options that would be activated via `-Dxxx` flags, this would be the first.

I think it might be more Emscripten-idiomatic to make this a `-s WARN_ON_SLOW_SIMD=0/1` option instead, which would in emcc.py internally enable the needed `-D` define. (and document that it is a compile time -s setting rather than a link time setting in src/settings.js and docs/emcc.txt). @sbc100 wdyt?

Reply:
Apart from that, the feature itself is a great idea!

Reply:
I'm happy do skip the setting and just go for a macro.. then the compiler doesn't need to be aware of it at all which is win in my book. 

I'm not sure about the naming though..  Why the two leading underscores but not two ending underscores?     The only other macro I can see that have in this vein is `__EMSCRIPTEN_TRACING__` but as @juj points out that is internally set by the compiler.    If this is designed to be set by the user maybe just `EMSCRIPTEN_SIMD_COMPAT_SLOW`?

Reply:
Open to suggestion on the names, I picked the __ prefix because I looked at the command line invocation of emcc and saw __EMSCRIPTEN_major. I can remove the __ prefix. And also go with your suggestion of changing WASM_SIMD_COMPAT_SLOW.

Reply:
And also leaving it as a macro.